integrations/transportation/ryanair.js

```diff
@@ -0,0 +1,138 @@
+import moment from 'moment';
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PLANE } from '../../definitions';
+
+const agent = request.agent();
+const BASE_URL = 'https://api.ryanair.com/userprofile/rest/api/v1/';
+const LOGIN_URL = `${BASE_URL}login`;
+const PROFILE_URL = `${BASE_URL}secure/users/`;
+// user info available at `${PROFILE_URL}${customerId}/profile/full/`
+
+async function logIn(username, password, logger) {
+  const res = await agent
+    .post(LOGIN_URL)
+    .type('application/x-www-form-urlencoded')
+    .send({
+      password,
+      username,
+    });
+
+  if (!res.ok) {
+    logger.logDebug('Error while logging in.');
+    throw Error('We\'re not logged in.');
+  }
+
+  return {
+    token: res.body.token,
+    customerId: res.body.customerId,
+  };
+}
+
+async function connect(requestLogin, requestWebView, logger) {
+  // Here we can request credentials etc..
+
+  // Here we can use two functions to invoke screens
+  // requestLogin() or requestWebView()
+  const { username, password } = await requestLogin();
+
+  if (!(password || '').length) {
+    throw Error('Password cannot be empty');
+  }
+
+  return logIn(username, password, logger);
+}
+
+function disconnect() {
+  // Here we should do any cleanup (deleting tokens etc..)
+  return {};
+}
+
+async function getAllFlights(bookings, customerId, token) {
+  const allFlights = [];
+
+  await Promise.all(bookings.map(async (entry) => {
+    await agent
```

Comment:
Needs to be wrapped in a try/catch
It should use a `HTTPError` now that #109 has been merged. You can look at some of the examples of that PR.